docs/development/design/embed-api.rst

```diff
@@ -104,36 +99,112 @@ Allow us to query the content of the section, with all links re-written as absol
          }
       }
 
-Notes
------
+Implemention
+------------
+
+If a section or page doesn't exist, we return 404.
+  This guarantees that the client requesting this resource has a way of knowing the response is correct.
 
-- If a section or page doesn't exist, we return 404.
-- All links are re-written to be absolute (this is already done).
-- All sections listed are from html tags that are linkeable, this is, they have an ``id``
+All links are re-written to be absolute.
+  Allow the content to be located in any page and in external sites
+  (this is already done).
+
+All sections listed are from html tags that are linkeable.
+  This is, they have an ``id``
   (we don't rely on the toctree from the fjson file anymore).
-- The IDs returned don't contain the redundant ``#`` symbol.
-- The content is an string with a well formed HTML block.
-- We could also support only ``url`` as argument for ``/sections`` and ``/pages``,
-  but this introduces another way of querying the API.
-  Having two ways of querying the API makes it *less-cacheable*.
-- Returning the extra js and css requires parsing the HTML page itself,
-  rather than only the content extracted from the fjson files (this is for sphinx).
-  We can use both, the html file and the json file, but we could also just start parsing the full html page
-  (we can re-use code from the search parsing to detect the main content).
-- ``extras`` could be returned only on ``/pages``, or only on ``/sections``.
+  This way is more easy to parse and get the wanted section,
+  instead of restricting to some types of contents.
+
+The IDs returned don't contain the redundant ``#`` symbol.
+  The fragment part could be used in external tools.
+
+The content is an string with a well formed HTML block.
+  Malformed HTML can cause the content to be rendered in unexpected ways.
+  Some HTML tags are required to be be inside other tags or be surrounded by other tags,
+  examples are ``li`` tags inside ``ul`` or ``dd`` tags inside ``dl`` and having a ``dt`` tag.
+
+  For example extracting the ``title`` section from this snipped:
+
+  .. code:: html
+
+     <dl>
+      ...
+
+      <dt id="foo">Foo</dt>
+      <dd>Some definition</dd>
+
+      <dt id="title">Title<dt>
+      <dd>Some definition</dd>
+
+      ...
+     </dl>
+
+  Would result in
+
+  .. code:: html
+
+     <dl>
+      <dt id="title">Title<dt>
+      <dd>Some definition</dd>
+     </dl>
+
+  Instead of
+
+  .. code:: html
+
+     <dd>Some definition</dd>
+
+  Note that we only try to keep the current structure,
+  if the page contains malformed HTML, we don't try to *fix it*.
+  This improvement can be shared with the current API (v2).
```

Comment:
We can't back-port this since it's a breaking change from a user's perspective: the render of the content in the tooltip will be completely different (see https://github.com/readthedocs/readthedocs.org/pull/8039#discussion_r603396695)